### PR TITLE
32 us as a user i want to find any type of channels

### PR DESCRIPTION
**Pull request by estelle-rcr**
# Description

Please provide a detailed description of what was done in this PR.
Precise the issue that you are resolving.

# Changes include

- [X] New feature (non-breaking change that adds functionality)
- [ ] Bugfix (non-breaking change that solves an issue)

# Breaking changes

Please complete this section if any breaking changes have been made, otherwise delete it.

# Checklist

- [X] I have assigned this PR to myself
- [X] I have added at least 1 reviewer
- [X] I have added the needed labels
- [X] I have linked this PR to an issue
- [X] I have linked this PR to a milestone
- [X] I have linked this PR to a project
- [X] I have tested this code
- [X] I have added / updated tests (end-to-end tests ...)
- [ ] I have updated the README and other relevant documents (guides...)
- [X] I have added sufficient documentation both in code, as well as in the READMEs

# Additional comments

J'ai jamais autant coché de cases :)))

### CODE REVIEW

**Comment by cnstll** [comment]
Ginormous PR :O

File: backend/app/src/channel/channel.controller.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    @GetCurrentUserId() userId: string,
+    @Param('id') channelId: string,
+    @Body() dto: ChannelDto,
+    @Res() res: Response,
+  ) {
+    return this.channelService.editChannelById(userId, channelId, dto, res);
+  }
+
+  // Delete channel
+  @Delete(':id')
```
[the remark this answers is not in the document]
Comment (cnstll) [reply backend/app/src/channel/channel.controller.ts:88]: will be handled in another issue

File: backend/app/src/channel/channel.controller.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+    return this.channelService.getChannelByUserId(userId, channelId);
+  }
+
+  @Get('get-users-of-a-channel/:id')
+  getUsersOfAChannel(@Param('id') channelId: string) {
+    return this.channelService.getUsersOfAChannel(channelId);
+  }
+
+  /* Is it possible without routing on id? */
+  @Get('get-role-user-channel/:id')
```
[the remark this answers is not in the document]
Comment (cnstll) [reply backend/app/src/channel/channel.controller.ts:58]: Look into prisma error handling

**Comment by estelle-rcr** [comment]
- Un seul try
- Rendre obligatoire name
- Check si channel non existant avant de demander un role ou les utilisateurs

**Pull request merged**